File: NodeJS_Server/routes/user_apis.mjs
import express from 'express';
import USER_FNS from '../controllers/users_controller.mjs';

const router = express.Router();

// Get all users
router.get('/', USER_FNS.getAllUsers);

// Get a single user by ID
router.get('/:id', USER_FNS.getUserById);

// Get a single user by username
router.get('/username/:username', USER_FNS.getUserByUsername);

// Create a new user
router.post('/', USER_FNS.createUser);

// Change password for a user
router.put('/change-password', USER_FNS.changePassword);

// Delete a user by ID
router.delete('/:id', USER_FNS.deleteUser);

export default router;

File: FastAPI_Server/routes/user_apis.py
from fastapi import APIRouter, Depends
from controllers.users import UserController
from models.user import UserCreate, UserResponse, ChangePassword
from typing import List

router = APIRouter()

def get_user_controller():
    return UserController()

@router.get("/", response_model=List[UserResponse])
async def get_all_users(controller: UserController = Depends(get_user_controller)):
    """Get all users"""
    return await controller.get_all_users()

@router.get("/{user_id}", response_model=UserResponse)
async def get_user_by_id(user_id: str, controller: UserController = Depends(get_user_controller)):
    """Get a single user by ID"""
    return await controller.get_user_by_id(user_id)

@router.get("/username/{username}", response_model=UserResponse)
async def get_user_by_username(username: str, controller: UserController = Depends(get_user_controller)):
    """Get a single user by username"""
    return await controller.get_user_by_username(username)

@router.post("/", response_model=UserResponse, status_code=201)
async def create_user(user_data: UserCreate, controller: UserController = Depends(get_user_controller)):
    """Create a new user"""
    return await controller.create_user(user_data)

@router.put("/change-password")
async def change_password(change_data: ChangePassword, controller: UserController = Depends(get_user_controller)):
    """Change password for a user"""
    return await controller.change_password(change_data)

@router.delete("/{user_id}")
async def delete_user(user_id: str, controller: UserController = Depends(get_user_controller)):
    """Delete a user by ID"""
    return await controller.delete_user(user_id)
